src/trusted_ca_keys/trusted_ca_keys.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <fstream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include "oslogin_utils.h"
+#include "trusted_ca_keys.h"
+
+using std::cout;
+using std::cerr;
+using std::endl;
+using std::flush;
+using std::string;
+using std::stringstream;
+using std::vector;
+using std::ofstream;
+
+using oslogin_utils::HttpGet;
+using oslogin_utils::ParseJsonToSuccess;
+using oslogin_utils::ParseJsonToEmail;
+using oslogin_utils::ParseJsonToSshKeys;
+using oslogin_utils::UrlEncode;
+using oslogin_utils::ParseJsonToCAKeys;
+
+ofstream
+CreateTmpKeysFile(const string& trusted_keys_filename, const string& trusted_keys_tmp_filename) {
+
+  ofstream trusted_keys_tmp_file(trusted_keys_tmp_filename.c_str());
+
+  if (!trusted_keys_tmp_file.is_open()) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Can't open '"
+            << trusted_keys_tmp_filename << "' for writing";
+    perror(err_msg.str().c_str());
```

Comment:
the use of `perror()` is not consistent with the code base, the use of `syslog()` is consistent across the code base and seems more appropriate.

Reply:
The "code based" has another use case. Again, I didn't have a goal to copy paste the existing behavior.
The existing code, for example `authorized_keys` tool, is invoked from `SSHD`. It simply cannot print any information for a user to stdout/stderr. That's why it uses syslog.

My tool serves in a totally diffeernt situation. It is executed from SystemD unit and it _has to_ print to stdout. Systemd sends _everything_ from its stdout/stderr to its own journal (read, syslog)
I really don't see any reason to _duplicate_ this information in the journal (syslog). it will be there anyway with the help of Systemd!